datamanagement/dlp_bcl_fastq_import.py

```diff
@@ -76,6 +84,76 @@ def _update_info(info, key, value):
         info[key] = value
 
 
+def check_fastqs(library_id, fastq_file_info):
+    logging.info("Checking if BCL2FASTQ generated complete set of fastqs.")
+
+    # Get indices from colossus
+    colossus_samples = query_colossus_dlp_cell_info(library_id)
+    colossus_index_sequences = set(sample["index_sequence"] for sample in colossus_samples.values())
+
+    # Get indices from given fastqs
+    fastq_index_sequences = set([fastq["index_sequence"] for fastq in fastq_file_info])
+    index_sequences = fastq_index_sequences
+
+    if len(colossus_index_sequences - fastq_index_sequences) != 0:
+        logging.info("BCL2FASTQ skipped indices {}".format(colossus_index_sequences - fastq_index_sequences))
+        index_sequences = colossus_index_sequences
+
+    fastqs_to_be_generated = dict()
+    threshold = 20
+
+    for index in index_sequences:
+        fastqs_containing_index = [fastq for fastq in fastq_file_info if fastq["index_sequence"]==index]
+        fastq_lane_numbers_to_be_generated = set(range(1,5))
+        for fastq in fastqs_containing_index:
+            sequence_lane = fastq["sequence_lanes"][0]
+            lane_number = sequence_lane["lane_number"]
+            fastq_lane_numbers_to_be_generated = fastq_lane_numbers_to_be_generated - set([lane_number])
+
+        fastqs_to_be_generated[index] = list(fastq_lane_numbers_to_be_generated)
+
+    return fastqs_to_be_generated
+
+
+def generate_empty_fastqs(output_dir, library_id, fastqs_to_be_generated):
+    # Ex: SA992-A90632-R54-C54_S317_L004_R1_001.fastq.gz
+
+    sublibraries = get_colossus_sublibraries_from_library_id(library_id)
+
+    index_sequence_map = {}
+    file_names = []
+    reads = [1,2]
+
+    for sublibrary in sublibraries:
+        index_sequence = sublibrary["primer_i7"] + "-" + sublibrary["primer_i5"]
+        sample_id = sublibrary["sample_id"]["sample_id"]
+        index_sequence_map[index_sequence] = {
+            "sample_id":    sample_id,
+            "row":          sublibrary["row"], 
+            "column" :      sublibrary["column"]
+        }    
+
+    for fastq_index in fastqs_to_be_generated.keys():
+        row = index_sequence_map[fastq_index]["row"]
+        column = index_sequence_map[fastq_index]["column"]
+        sample_id = index_sequence_map[fastq_index]["sample_id"]
+        
+        samplename = "-".join([sample_id,library_id,"R{}".format(row), "C{}".format(column)])
+        for lane_num in fastqs_to_be_generated[fastq_index]:
+            for read in reads:
+                filename =  "_".join([samplename, "S0EMPTY", "L00{}".format(lane_num), "R{}".format(read), "001.fastq.gz"])
+                file_names.append(filename)
+
+    for filename in file_names:
+        filepath = os.path.join(output_dir, filename)
+        if not os.path.exists(filepath):
+            logging.info("Creating empty file {} at {}.".format(filename, filepath))
+            with open(filepath, 'wb') as f:
+                f.write("")
```

Comment:
This will generate an empty file, but not a valid gzip file.  Use gzip.open instead of open.